src/solvers/dgmulti/types.jl

```diff
@@ -76,25 +76,50 @@ function DGMulti(element_type::AbstractElemShape,
   return DG(rd, nothing #= mortar =#, surface_integral, volume_integral)
 end
 
-
-# now that DGMulti is defined, we can define constructors for VertexMappedMesh which use dg::DGMulti
+# now that `DGMulti` is defined, we can define constructors for `DGMultiMesh` which use `dg::DGMulti`
 """
-    VertexMappedMesh(vertex_coordinates, EToV, dg::DGMulti;
-                     is_on_boundary = nothing,
-                     is_periodic::NTuple{NDIMS, Bool} = ntuple(_->false, NDIMS)) where {NDIMS, Tv}
+  DGMultiMesh(vertex_coordinates, EToV, dg::DGMulti;
+              is_on_boundary = nothing,
+              is_periodic::NTuple{NDIMS, Bool} = ntuple(_->false, NDIMS)) where {NDIMS, Tv}
 
 Constructor which uses `dg::DGMulti` instead of `rd::RefElemData`.
 """
-VertexMappedMesh(vertex_coordinates, EToV, dg::DGMulti; kwargs...) =
-  VertexMappedMesh(vertex_coordinates, EToV, dg.basis; kwargs...)
+DGMultiMesh(vertex_coordinates, EToV, dg::DGMulti; kwargs...) =
+  DGMultiMesh(vertex_coordinates, EToV, dg.basis; kwargs...)
 
 """
-    VertexMappedMesh(triangulateIO, dg::DGMulti, boundary_dict::Dict{Symbol, Int})
+  DGMultiMesh(triangulateIO, dg::DGMulti, boundary_dict::Dict{Symbol, Int})
 
 Constructor which uses `dg::DGMulti` instead of `rd::RefElemData`.
 """
-VertexMappedMesh(triangulateIO, dg::DGMulti, boundary_dict::Dict{Symbol, Int}) =
-  VertexMappedMesh(triangulateIO, dg.basis, boundary_dict)
+DGMultiMesh(triangulateIO, dg::DGMulti, boundary_dict::Dict{Symbol, Int}) =
+  DGMultiMesh(triangulateIO, dg.basis, boundary_dict)
+
+"""
+  CartesianMesh(dg::DGMulti; cells_per_dimension,
```

Comment:
What's the motivation for introducing a new name `CartesianMesh` instead of making it just another `DGMultiMesh` constructor? IIRC, we do something similar for the `StructuredMesh` or `P4estMesh`, which can also be constructed as a Cartesian mesh. If I remember correctly, `CartesianMesh` is only used for a convenience ctor and not for dispatch, so there doesn't seem to be "hard" motivation for a new type. I might be mistaken though...

Reply:
Sounds reasonable to use the same name (`DGMultiMesh`) if there won't be another underlying type :+1:

Reply:
I introduced `CartesianMesh` mainly to match https://github.com/trixi-framework/Trixi.jl/pull/1010. I'm fine with swapping it for a `DGMultiMesh` constructor. 

I could use some feedback though - I will need to dispatch on the mesh type eventually. I can do this by adding type parameters to `DGMultiMesh` (e.g., `DGMultiMesh{NDIMS, ..., Cartesian}` or `DGMultiMesh{NDIMS, ..., Curved}`) or just by introducing new mesh types and making `DGMultiMesh` an abstract type (e.g., `CartesianMesh <: AbstractDGMultiMesh`). Any preference?

Reply:
I don't have a strong opinion on this

Reply:
There's to little for me to go on, to make a confident suggestion. For this PR, I'd stick with not creating a separate type.

For the future, I would probably use an explicit type rather than just a type parameter. This allows you to choose different (more efficient/cleaner?) implementations for simpler mesh types (such as Cartesian) and makes it easier to read (and write) for dispatch. But I'd leave this discussion for the PR in which you actually have to introduce them ;-)

Reply:
OK, this sounds good. I'll save the introduction of a new type for later, but I think I should then rename some of the variables to emphasize what assumptions existing routines are making based on `DGMulti` mesh types. This should just be a renaming 

1. `AbstractMeshData` -> `AbstractDGMultiMesh` and `DGMultiMesh` -> `AffineMesh` 
2. add `AffineMesh` type annotations whenever I assume constant geometric terms.

I'll then introduce `CurvedMesh <: AbstractDGMultiMesh` in a future PR.